src/views/Auth/Login.jsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import { signInUser, signUpUser } from '../../services/user';
+import { useHistory } from 'react-router-dom';
+import './Login.css';
+import { useUserContext } from '../../context/UserContext';
+
+export default function Login() {
+  const history = useHistory();
+  const { setCurrentUser } = useUserContext();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [type, setType] = useState('sign-in');
+  const [error, setError] = useState('');
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      if (type === 'sign-in') {
+        const data = await signInUser({email, password});
+        setCurrentUser(data);
+        history.push('/');
+      } else {
+        const data = await signUpUser({email, password});
+        setCurrentUser(data);
+          history.push('/');
+      }
+    } catch (error) {
+      setError(e.message);
+    }
+  };
+
+  return (
+    <div className="button">
+      <h1>
+        <span
+          className={type === 'sign-in' ? 'active' : ''}
+          onClick={() => setType('sign-in')}
+        >
+          Sign In
+        </span>
+        <span
+          className={type === 'sign-up' ? 'active' : ''}
+          onClick={() => setType('sign-up')}
+        >
+          Sign Up
+        </span>
```

Comment:
When you're signing in or up it's difficult to know what you're doing. It would be a better user experience if the button or text changed to show the user what was happening. Adding a message here as well that stated when the user is already signed up would be a great feature.